app/controllers/posts_controller.rb

```diff
@@ -7,6 +7,25 @@ def index
   def show
     @post = Post.find(params[:id])
     @user = User.find(@post.author_id)
-    @comments = Comment.where(posts_id: @post.id)
+    @comments = Comment.where(post_id: @post.id)
+    @comment = Comment.all.where("post_id = #{params[:id]}, user_id = #{params[:user_id]}")
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+
+    @post = @user.posts.new(post_params)
+    if @post.save
+      redirect_to user_path(@user), notice: 'Successfully added a Post!'
+    else
+      flash.now[:notice] = 'Failed to create a Post.'
+      render :new
```

Comment:
- [ ] I would recommend adding a `new` method to create an instance.

```ruby
class PostsController < ApplicationController

    ...

    def new
      @post = Post.new
    end
end
```

Reply:
Thanks, @Dipeshtwis for your kind suggestion and feedback!. This suggestion is implemented on commit 54ec112.